x-pack/plugins/alerting/server/rules_client/lib/siem_legacy_actions/format_legacy_actions.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { chunk } from 'lodash';
+import type { SavedObjectsFindOptionsReference, Logger } from '@kbn/core/server';
+import pMap from 'p-map';
+import { RuleAction, Rule } from '../../../types';
+import type { RuleExecutorServices } from '../../..';
+import { injectReferencesIntoActions } from '../../common';
+import { transformToNotifyWhen } from './transform_to_notify_when';
+import { transformFromLegacyActions } from './transform_legacy_actions';
+import { LegacyIRuleActionsAttributes, legacyRuleActionsSavedObjectType } from './types';
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+interface LegacyGetBulkRuleActionsSavedObject {
+  alertIds: string[];
+  savedObjectsClient: RuleExecutorServices['savedObjectsClient'];
+  logger: Logger;
+}
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+export interface LegacyActionsObj {
+  ruleThrottle: string | null;
+  legacyRuleActions: RuleAction[];
+}
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+export const legacyGetBulkRuleActionsSavedObject = async ({
+  alertIds,
+  savedObjectsClient,
+  logger,
+}: LegacyGetBulkRuleActionsSavedObject): Promise<Record<string, LegacyActionsObj>> => {
+  const references = alertIds.map<SavedObjectsFindOptionsReference>((alertId) => ({
+    id: alertId,
+    type: 'alert',
+  }));
+  const errors: unknown[] = [];
+  const results = await pMap(
+    chunk(references, 1000),
+    async (referencesChunk) => {
+      try {
+        return savedObjectsClient.find<LegacyIRuleActionsAttributes>({
+          type: legacyRuleActionsSavedObjectType,
+          perPage: 10000,
+          hasReference: referencesChunk,
+        });
+      } catch (error) {
+        logger.error(
+          `Error fetching rule actions: ${error instanceof Error ? error.message : String(error)}`
+        );
+        errors.push(error);
+        return [];
+      }
+    },
+    { concurrency: 1 }
+  );
+  const actionSavedObjects = results.flat().flatMap((r) => r.saved_objects);
+
+  if (errors.length) {
+    throw new AggregateError(errors, 'Error fetching rule actions');
```

Comment:
why are we using `AggregateError`?

Reply:
this is part of the original Security SOlution code, that migrated actions from https://github.com/elastic/kibana/blob/4072cf2d3d83995881ced99c4ef900a853813b33/x-pack/plugins/security_solution/server/lib/detection_engine/rule_actions_legacy/logic/rule_actions/legacy_get_bulk_rule_actions_saved_object.ts#L60-L62

I think, because there are multiple errors, `AggregateError` is used for its purpose